src/Microsoft.Framework.Runtime/Servicing/Breadcrumbs.cs

```diff
@@ -37,36 +45,55 @@ public static bool IsPackageServiceable(PackageInfo package)
             return false;
         }
 
-        public void CreateBreadcrumb(string packageId, SemanticVersion packageVersion)
+        public void AddBreadcrumb(string packageId, SemanticVersion packageVersion)
         {
-            if (string.IsNullOrWhiteSpace(_breadcrumbsFolder))
+            _breadcrumbsToWrite.Add(new BreadcrumbInfo()
             {
-                return;
-            }
+                PackageId = packageId,
+                PackageVersion = packageVersion
+            });
+        }
 
-            // Create both files for now until we get clear instructions about the format of the name
-            CreateBreadcrumbFile(packageId);
-            CreateBreadcrumbFile(packageId + "." + packageVersion);
+        public void WriteAllBreadcrumbs(bool background = false)
+        {
+            if (background)
+            {
+                ThreadPool.QueueUserWorkItem(_ =>
```

Comment:
Add an environment variable that makes this NOOP. I want to disable this when I'm running performance tests.